feat(ui-kit): Quiz component is added with UI , storybook and tests.

manishRise11 opened the pull request:
## Description

<!-- Provide a clear and concise description of the changes made in this pull request for UI-Kit components. -->

- Component name: `QuizComponent`
- Figma link or screenshot: [Figma Design](https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=7224-259576&t=KkMB8R0lxAuyJaA2-4)

![Image](https://github.com/user-attachments/assets/f195256e-bdef-4a9f-b4d6-a7dbe5c32451)
---


## React Best Practices Checklist

- [ ] I have avoided **unnecessary re-renders** by using composition (`children` as props), whenever possible. This applies, in particular, to composite components (like lists of components, and the like).
- [ ] **I have designed my components as server-side whenever possible**—I have not used `useState`, `useEffect`, `useContext`, or any other React hook that works with "use client" without a good reason.
- [ ] **I have not put props in state**—in general, props should remain as props unless the component needs an `initialValue`, and judiciously manage its own independent state detached from the parent component.
    - If I have stored props in state, I have explained where exactly and justified it with a solid reason here below:
    
- [ ] **I have avoided using Context** in any critical component in terms of performance
    - If I have used Context, I have explained where exactly and justified it with a solid reason here below:


## General Checklist

- [ ] I have run the lint, build, and test commands locally, as explained in the README, before creating this PR to ensure the code is clean and functional.
- [ ] My changes align with the **core models** that have been discussed and agreed upon.
- [ ] The component design has been compared to the **Figma design** to ensure consistency.
- [ ] The **component is fully responsive** and behaves as expected across different screen sizes.
- [ ] The component **supports keyboard web accessibility** (if applicable).
- [ ] I have written **unit tests** or **integration tests** to cover the component's functionality above the coverage threshold.
- [ ] I have updated the **documentation** for the component, including usage examples and guidelines.
- [ ] I have **commented** my code, particularly in complex or non-obvious areas.
- [ ] The **component is properly exported** for use in other parts of the application.




---

## Related Issues

<!-- Link any related issue or task. For example: -->
Fixes #171

manishRise11 commented:
Hi @alebg sir,

- In the preview/student view, callbacks are only triggered when the user clicks a button (not on every change).
- 
- All hard-coded values have been removed from the main components; simulation logic is now in Storybook.
- 
- For file upload logic: Now, instead of handling file logic inside the component, I'm passing callbacks (e.g., onFilesChange, onFileDelete, onFileDownload) from the parent/Storybook. The component receives a setFile callback to update its file state.
- Please confirm if this approach aligns with your expectations or if you recommend any further adjustments.

Thanks!

manishRise11 commented:
Hi @carlapbg mam,

- The error when switching to quiz type 2 is fixed by updating the type-change logic and ensuring correct data is passed.
- For square, icon-only badges, I’m using the existing leftIcon/rightIcon prop and applying a square style with className. Please let me know if this is fine or if you prefer a dedicated prop/variant.
- Changed all quiz preview titles to h5 with text-xl, and descriptions to text-lg as requested.

alebg commented:
@manishRise11 in the future please don't copy the issue description here, no need to have that content twice. Use the PR template we've provided instead

manishRise11 commented:
> Thanks! Just one thing: In the Quiz Type 1, student view, when I click in “show solution” it still shows me the rectangular badge.
> 
> <img width="560" alt="Screenshot 2025-05-22 at 16 11 18" src="https://github.com/user-attachments/assets/7b5bcbd7-5cc5-4da5-bc02-de919e7e1986" />
> 
> 

Ok mam, I will change this badge to square

manishRise11 commented:
@carlapbg Ma’am, I have run this locally and observed that the badges appear as squares, as shown in the attached image. I have also tested this in different browsers, and the badges consistently appear square in all of them.
<img width="566" alt="Screenshot 2025-05-23 at 7 50 53 PM" src="https://github.com/user-attachments/assets/d668645a-5123-4009-9acb-a5496d6abeed" />

alebg commented:
@manishRise11 the image handling doesn't make much sense in my opinion, in relation to the uploader.

I see that if there's an image that already is present in the back (which in your code we know because we're passing an id + url), then you're initializing a new File object for the uploader component with a hardcoded name "image.jpg" and a hardcoded type.

It would be better to pack types for files with everything we need there: id, name, size, url, fileType, type, and thumbnailUrl for the media (image ,video) types

![immagine](https://github.com/user-attachments/assets/48f9a123-f0be-4c4c-bcae-ceaea9f23dca)

Then ideally we would just pass this `image` data to the uploader component and let that component handle it. I think part of the problem comes from that component itself, where the `UploadedFileType` demands things that you will not get from the backend, like a `File` object. I didn't think about this one through in that PR, but whenever you already have a file uploaded, what you will get is a signed URL that will allow you to download that file on demand. We won't "pre-download" this as file objects before instantiating the uploader component.

So just to say, it seems to me that the file uploader component needs some refactoring such that all of this is consistent all around the app, and use something more similar to my `...Metadata` draft type above. I would do it myself, but this will touch open PRs, so it's better if we discuss it together

manishRise11 commented:
- Resolved the issue reported by @carlapbg — In Quiz Type 1, the student view no longer shows the rectangular badge when “Show Solution” is clicked.
- Fixed merge conflicts and updated documentation for the CourseBuilderType.
- As suggested by Jatin , I’ve refactored the enum structure into a single unified export:

```
export enum CourseElementType {
  LessonNote = "lessonNote",
  CoachingSession = "coachingSession",
  Quiz = "quiz",
}

```

@alebg Please let me know if this structure aligns with what you expect or if there’s anything I should revise.

alebg commented:
closed in favor of #248
